Migrate from Zowe CLI package to Zowe SDKs

t1m0thyj opened the pull request:
## Proposed changes

<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. -->

Resolves #2719

Had to temporarily add an override for `@zowe/imperative` in FTP extension's package.json, because we don't yet have a vNext version of the FTP plug-in with an Imperative vNext peer dependency.

Also adds a build script for updating Zowe SDKs: `pnpm run update-sdks`. In the future when v3 goes GA, we'll want to update the tag in this script from `next` to `zowe-v3-lts` 😋 

## Release Notes

<!-- Include the Milestone Number and a small description of your change that will be added to the changelog -->
<!-- If there is a linked issue, it should have the same milestone as this PR -->

Milestone: 3.0.0

Changelog: Migrated from `@zowe/cli` package to individual Zowe SDK packages.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

t1m0thyj commented:
Since this PR introduces a lot more `@zowe`-scoped dependencies into package.json, would it be helpful to have a script in the repo that we could run to update them all to the latest version?

JillieBeanSim reviewed (approved):
Thanks @t1m0thyj LGTM!

t1m0thyj commented:
Build is currently failing because we use a method marked as internal in Imperative, waiting on https://github.com/zowe/zowe-cli/pull/2068 to be published

zFernand0 reviewed (dismissed):
LGTM! 😋

traeok commented on packages/zowe-explorer-api/__tests__/__unit__/profiles/ZoweExplorerZosmfApi.unit.test.ts:
```diff
@@ -9,8 +9,14 @@
-import * as zowe from "@zowe/cli";
-import { IZosmfInfoResponse } from "@zowe/cli";
+import { Login, Logout } from "@zowe/core-for-zowe-sdk";
+import * as imperative from "@zowe/imperative";
+import * as zosconsole from "@zowe/zos-console-for-zowe-sdk";
+import * as zosfiles from "@zowe/zos-files-for-zowe-sdk";
+import * as zosjobs from "@zowe/zos-jobs-for-zowe-sdk";
+import * as zostso from "@zowe/zos-tso-for-zowe-sdk";
+import * as zosuss from "@zowe/zos-uss-for-zowe-sdk";
+import * as zosmf from "@zowe/zosmf-for-zowe-sdk";
```
We might want to evaluate the amount of `import *` statements in production code. Nothing here to worry about as this is test code - was mostly used as an example - but we might want to check if this is done elsewhere in ZE API's source to maintain runtime efficiency :yum: 

Granted, since we use a bundler this really doesn't apply to us, but it might help improve performance of non-Webpacked extenders.

traeok reviewed (dismissed):
LGTM, thanks @t1m0thyj :tada:

zFernand0 commented on package.json:
```diff
@@ -76,6 +75,7 @@
     "posttest:integration": "ts-node ./scripts/clean-env.ts",
-    "preinstall": "npx only-allow pnpm"
+    "preinstall": "npx only-allow pnpm",
+    "update-sdks": "pnpm -r update \"@zowe/*@next\" \"!@zowe/zowe-explorer-api\""
```
❤️

zFernand0 reviewed (approved):
LGTM! 😋

zFernand0 commented:
This shall be merged last 😋

JillieBeanSim reviewed (changes requested):
when I run `pnpm run build` on this branch it creates/updates 3 files with licenses but files have nothing else in them. Should they be removed or updated as seen?
<img width="1395" alt="Screenshot 2024-02-28 at 11 53 53 AM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/49491949/c19d65ed-998b-4036-a3de-77bf2a1b3855">

t1m0thyj commented:
> when I run `pnpm run build` on this branch it creates/updates 3 files with licenses but files have nothing else in them. Should they be removed or updated as seen? <img alt="Screenshot 2024-02-28 at 11 53 53 AM" width="1395" src="https://private-user-images.githubusercontent.com/49491949/308644220-c19d65ed-998b-4036-a3de-77bf2a1b3855.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNDM5OTEsIm5iZiI6MTcwOTE0MzY5MSwicGF0aCI6Ii80OTQ5MTk0OS8zMDg2NDQyMjAtYzE5ZDY1ZWQtOTk4Yi00MDM2LWEzZGUtNzdiZjJhMWIzODU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDE4MDgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2MzUyNDZlMmNmMjVmZTA3OTRhMzNkYzU2YTVmNDc0NWRhOWNlYmFhZTZjZjZjMGJhMWYyYjYzZjEyMGJmNTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7rnLClYyPnCKOYxrOqBLRKHKbavoVQRNBqB3Ysgb8xI">

@JillieBeanSim Thanks for catching this! I think what was happening was `pnpm license` added a trailing line and then `pnpm pretty` removed it 😆 The empty module mocks were kind of weird, so I removed them and updated tests to work without them.

JillieBeanSim reviewed (approved):
LGTM! thanks @t1m0thyj

traeok reviewed (approved):
Thanks @t1m0thyj for updating the newly-merged changes to use the SDKs 😁
